src/cli/list.rs

```diff
@@ -0,0 +1,283 @@
+use std::path::PathBuf;
+
+use clap::Parser;
+use comfy_table::presets::NOTHING;
+use comfy_table::{Attribute, Cell, Color, ContentArrangement, Table};
+use human_bytes::human_bytes;
+use rattler_conda_types::Platform;
+use rattler_lock::{LockedDependency, LockedDependencyKind};
+use serde::Serialize;
+
+use crate::lock_file::load_lock_file;
+use crate::project::SpecType;
+use crate::Project;
+
+// an enum to sort by size or name
+#[derive(clap::ValueEnum, Clone, Debug, Serialize)]
+pub enum SortBy {
+    Size,
+    Name,
+}
+
+/// List project's packages. Highlighted packages are explicit dependencies.
+#[derive(Debug, Parser)]
+#[clap(arg_required_else_help = false)]
+pub struct Args {
+    /// List only packages matching a regular expression
+    #[arg()]
+    pub regex: Option<String>,
+
+    /// The platform to list packages for. Defaults to the current platform.
+    #[arg(long)]
+    pub platform: Option<Platform>,
+
+    /// Whether to output in json format
+    #[arg(long)]
+    pub json: bool,
+
+    /// Whether to output in pretty json format
+    #[arg(long)]
+    pub json_pretty: bool,
+
+    /// Sorting strategy
+    #[arg(long, default_value = "name", value_enum)]
+    pub sort_by: SortBy,
+
+    /// The path to 'pixi.toml'
+    #[arg(long)]
+    pub manifest_path: Option<PathBuf>,
+}
+
+#[derive(Serialize)]
+struct PackageToOutput {
+    name: String,
+    version: String,
+    build: Option<String>,
+    size_bytes: Option<u64>,
+    kind: String,
+    channel: Option<String>,
+    is_explicit: bool,
+}
+
+pub async fn execute(args: Args) -> miette::Result<()> {
+    // Load the project
+    let project = Project::load_or_else_discover(args.manifest_path.as_deref())
+        .ok()
+        .ok_or_else(|| miette::miette!("No project found"))?;
+
+    // Load the platform
+    let platform = args.platform.unwrap_or_else(Platform::current);
+
+    // Load the environment
+    // NOTE(hadim): make it an argument once environments are implemented.
```

Comment:
```suggestion
    // TODO: NOTE(hadim): make it an argument once environments are implemented.
```
I would make this a TODO as we have a lot of todos added regarding the default env work and it will help our IDE tip us!